File: C/bogosort/chatGPT/chatGPT.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#define ARRAY_SIZE 1000

// Function to check if an array is sorted
int is_sorted(int *array, int size) {
    for (int i = 0; i < size - 1; i++) {
        if (array[i] > array[i + 1]) {
            return 0;
        }
    }
    return 1;
}

// Function to shuffle the array randomly
void shuffle(int *array, int size) {
    for (int i = 0; i < size; i++) {
        int j = rand() % size;
        int temp = array[i];
        array[i] = array[j];
        array[j] = temp;
    }
}

// Bogosort algorithm
void bogosort(int *array, int size) {
    srand(time(NULL));
    while (!is_sorted(array, size)) {
        shuffle(array, size);
    }
}

int main() {
    const int SIZE = 10;
    int arr[SIZE];


    // Fill array with random values
    for (int i = 0; i < SIZE; i++) {
        arr[i] = rand() % 1000; // Random values between 0 and 9999
    }

    // Sort the array using bogosort
    bogosort(arr, SIZE);

    return 0;
}

File: Java/bogosort/ChatGPT/ChatGPT.java
import java.util.Random;

public class ChatGPT {
    
    // Method to check if the array is sorted
    public static boolean isSorted(int[] array) {
        for (int i = 1; i < array.length; i++) {
            if (array[i - 1] > array[i]) {
                return false;
            }
        }
        return true;
    }

    // Method to shuffle the array randomly
    public static void shuffle(int[] array) {
        Random rand = new Random();
        for (int i = 0; i < array.length; i++) {
            int randomIndex = rand.nextInt(array.length);
            int temp = array[i];
            array[i] = array[randomIndex];
            array[randomIndex] = temp;
        }
    }

    // Bogosort algorithm
    public static void bogosort(int[] array) {
        while (!isSorted(array)) {
            shuffle(array);
        }
    }

    // Default main method to test Bogosort with 10 values
    public static void main(String[] args) {
        int size = 10;  // Size of the array
        int[] array = new int[size];
        
        // Fill the array with random values
        Random rand = new Random();
        for (int i = 0; i < size; i++) {
            array[i] = rand.nextInt(1000);  // Random values between 0 and 999
        }

        // Apply Bogosort to the array
        bogosort(array);
    }
}
